irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -404,5 +404,78 @@ namespace iroha {
       };
       return makeCommandResult(std::move(result), message_gen);
     }
+
+    WsvCommandResult PostgresWsvCommand::addAssetQuantity(
+        const shared_model::interface::types::AccountIdType &account_id,
+        const shared_model::interface::types::AssetIdType &asset_id,
+        const std::string &amount,
+        const int precision) {
+      std::string query =
+          (boost::format(
+              // clang-format off
+              R"(
+          WITH has_account AS (SELECT account_id FROM account WHERE account_id = '%s' LIMIT 1),
+               has_asset AS (SELECT asset_id FROM asset WHERE asset_id = '%s' AND precision = %d LIMIT 1),
+               amount AS (SELECT amount FROM account_has_asset WHERE asset_id = '%s' LIMIT 1),
```

Comment:
`SELECT amount FROM account_has_asset WHERE asset_id = '%s' LIMIT 1` will return the first account from the table which has the specified asset. It seems like constraint or account id is missing.